RFCs/0009-Fn::For.md

```diff
@@ -0,0 +1,903 @@
+                "Type": "AWS::EC2::Instance",
+                "Properties": {
+                  "InstanceType": "m1.small",
+                  "ImageId": {"Ref": "value"}
+                }
+              },
+              "key": {"Fn::Sub": "Instance${index}"}
+            }
+          }
+        ]
+    },
+    "Outputs": {
+        "SecondInstanceId": {
+            "Description": "Instance Id for Instance1",
+            "Value": { "Fn::Ref": "Instance1" }
+        },
+        "SecondPrivateIp": {
+            "Description": "Private ip for Instance1",
+            "Value": {
+                "Fn::GetAtt": [ "Instance1", "PrivateIp" ]
+            }
+        }
+    }
+}
+```
+* In Yaml
+```yaml
+AWSTemplateFormatVersion: "2010-09-09"
+Description: EC2 Instances with different AMIs
+Resources:
+  Fn::Merge:
+    - Fn::Map:
+        collection:
+          - ami-1
+          - ami-2
+          - ami-3
+        fragment:
+          Type: AWS::EC2::Instance
+          Properties:
+            InstanceType: m1.small
+            ImageId: !Ref value
+        key:
+          Fn::Sub: Instance${index}
+Outputs:
+  SecondInstanceId:
+    Description: Instance Id for Instance1
+    Value:
+      Fn::Ref: Instance1
+  SecondPrivateIp:
+    Description: Private ip for Instance1
+    Value:
+      Fn::GetAtt:
+        - Instance1
+        - PrivateIp
+
+```
+* Above templates are equivalent to a below template in Yaml
+```yaml
+AWSTemplateFormatVersion: '2010-09-09'
+Description: "AutoScaling LaunchConfiguration"
+Parameters:
+  AmiIds:
+    Type: CommaDelimitedList
+    Default: "ami-1,ami-2,ami-3"
+Resources:
+  Instance0:
+    Type: AWS::EC2::Instance
+    Properties:
+      InstanceType: "m1.small"
+      ImageId: "ami-1"
+  Instance1:
+    Type: AWS::EC2::Instance
+    Properties:
+      InstanceType: "m1.small"
+      ImageId: "ami-2"
+  Instance2:
+    Type: AWS::EC2::Instance
+    Properties:
+      InstanceType: "m1.small"
+      ImageId: "ami-3"
+Outputs:
+  SecondInstanceId:
+    Description: "Instance Id for Instance1"
+    Value: !Ref Instance1
+  SecondPrivateIp:
+    Description: "Private ip for Instance1"
+    Value: !GetAtt Instance1.PrivateIp
+```
+
+# Properties of `Fn::Map`
+
+#### Reference source list from Parameters
+* You can reference a source list from Parameters
+```yaml
+AWSTemplateFormatVersion: '2010-09-09'
+Description: "EC2 Instances with different AMIs"
+Parameters:
+  AmiIds:
+    Type: CommaDelimitedList
+    Default: "ami-1,ami-2,ami-3"
+Resources:
+  Fn::Merge:
+    - Fn::Map:
+        collection: !Ref AmiIds
+        fragment:
+          Type: AWS::EC2::Instance
+          Properties:
+            InstanceType: m1.small
+            ImageId: !Ref value
+        key:
+          Fn::Sub: Instance${index}
+  Bucket:
+    Type: AWS::S3::Bucket
+    Properties:
+      Name: "bucket"
+```
+
+#### Logical Id Customization
+* You are required to customize Names (Logical Ids) of Resource and Outputs using Index
```

Comment:
this should be "You are required to customize Names (Logical Ids) of Resource and Outputs using **Key**", right?